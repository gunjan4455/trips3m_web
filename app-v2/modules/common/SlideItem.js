import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';

const MediaLogo = styled.div({
  width: '120px',
  position: 'relative',
  height: '80px',
  overflow: 'hidden',
  margin: '15px auto 0',
});

const MediaBox = styled.div({
  maxWidth: '300px',
  textAlign: 'center',
  whiteSpace: 'normal',
});

const GStoryTextP = styled.p({
  maxHeight: '36px',
  overflow: 'hidden',
});

/**
* Img component is not used as it applies min-width and min-height to 100.
* This is not desired for logo and hence using HTML img tag instead.
*/
const MediaLogoImg = styled.img({
  display: 'block',
  maxWidth: '100%',
  margin: '0 auto 48px',
});

const SlideItem = ({ newsHeading, month, year, logoPath, imageAlt }) => {
  const datePublished = `${month} ${year}`;
  return (
    <MediaBox>
      <GStoryTextP className="f14 m0">{newsHeading}</GStoryTextP>
      <p className="f10 fw9 pt5 pfc4 m0">{datePublished}</p>
      <MediaLogo>
        <MediaLogoImg src={`${logoPath}?tr=w-140,h-40,c-at_max`} alt={imageAlt} className="absolute-center" />
      </MediaLogo>
    </MediaBox>
  );
};

SlideItem.propTypes = {
  logoPath: PropTypes.string.isRequired,
  month: PropTypes.string.isRequired,
  year: PropTypes.number.isRequired,
  newsHeading: PropTypes.string.isRequired,
  imageAlt: PropTypes.string.isRequired,
};

export default SlideItem;
