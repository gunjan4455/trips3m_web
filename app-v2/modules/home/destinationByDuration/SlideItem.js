import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Sensor from 'react-visibility-sensor';

import { parsePrice, capitalizeFirstLetter } from 'helpers/parsers';
import { DESTINATION_VIEWED, trackSegment, ORGANISM_CLICKED } from 'actions/segmentEvents';

import Img from 'components/Common/Img';

const SlideItem = ({ imagePath, imagekitUrl, name, packageCount, startingPrice, link, id }) => {
  let isComponentViewed = false;
  const onSensorChange = (isVisible) => {
    if (!isComponentViewed && isVisible) {
      isComponentViewed = isVisible;
      trackSegment({
        event: DESTINATION_VIEWED,
        category: `Destination:${name}`,
        section: 'Best Seller Destinations',
        object: `Destination/${id}`,
        cta: '',
        destination_id: id,
        value: startingPrice,
      });
    }
  };

  const trackSegmentEvent = () =>{
    trackSegment({
      event: ORGANISM_CLICKED,
      category: `HOME PAGE`,
      section: 'Explore Bestselling Packages',
      object: `Package Card`,
      cta: capitalizeFirstLetter(name)
    });
  };

  const renderChild = () => (
    <>
      <div className="radius4 overflowh" style={{ width: '120px', height: '150px' }}>
        <Img ikSrc={imagekitUrl} src={imagePath} alt={name} width={120} height={150} />
      </div>
      <Sensor onChange={onSensorChange} intervalDelay={3000} />
      <p className="f14p fw9 pfc3 mt8" style={{ width: '120px' }}>{capitalizeFirstLetter(name)}</p>
      <p className="f10 pfc3 mt4">Starting at: {parsePrice({ price: startingPrice, appendOnly: false })}</p>
    </>
  );

  return (
    <div>
      {
        __SERVER__ ?
          <a
            href={link}
            onClick={() => trackSegmentEvent()}
          >
            { renderChild() }
          </a> :
          <Link
            to={link}
            onClick={() => trackSegmentEvent()}
          >
            { renderChild() }
          </Link>
      }
    </div>
  );
};

SlideItem.propTypes = {
  link: PropTypes.string.isRequired,
  imagePath: PropTypes.string.isRequired,
  imagekitUrl: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  packageCount: PropTypes.number.isRequired,
  startingPrice: PropTypes.number.isRequired,
  id: PropTypes.number.isRequired,
  bestTimeToVisit: PropTypes.array.isRequired
};

export default SlideItem;
