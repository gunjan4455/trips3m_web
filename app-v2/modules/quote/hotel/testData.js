export const hotelInfo = {
  quoteCityId: 35268058,
  hotelId: 3436,
  tripDayId: 38054617,
  nights: '2',
  city: 'Calangute',
  name: 'Sunkissed Plaza',
  rating: 3,
  address: 'Calangute -  North Goa, Gauravaddo Calangute\r\nBardez GOA 403516 INDIA',
  taUrl: 'http://www.tripadvisor.com/Hotel_Review-g306995-d3916311-Reviews-Sunkissed_Plaza-Calangute_Goa.html',
  dId: 3916311,
  gId: 306995,
  roomType: ' Deluxe ',
  isAlternate: 0,
  comments: '',
  selectionStatus: null,
  description: 'Sunkissed Plaza, Goa, is set in a pleasing location near the blue waters of Calangute Beach. Situated far from the hustle-bustle of the city centre, this hotel offers a stay that is indeed tranquil. Guests can enjoy the beauty of the surroundings from their rooms with the gamut of amenities during the stay. \r\n\r\n<b>Location:</b>\r\nSunkissed Plaza is located at Gauravaddo Calangute Bardez. The Calangute Beach (Approx. 1km), and St Anthony\'s Chapel, Goa (Approx. 1km) are the nearby places of tourist interest from Sunkissed Plaza. Heavily fringed with coconut trees, Calangute Beach offers a spectacular view of the sea. Ideal for numerous water sports and beach fun, Calangute Beach, is a popular tourist destination near Sunkissed Plaza. Ancestral Goa Museum, Carambolim Lake, and Arvalam Waterfalls are the other places that are worth-exploring in Goa.\r\n\r\nDistance from Goa International Airport: 39 km (approx.)\r\nDistance from Thivim Railway Station: 22 km (approx.)\r\n\r\n<b>Hotel Features:</b>\r\nSunkissed Plaza is a ravishing estate surrounded by lush green trees. Simplistic yet comfy, Sunkissed Plaza features well-appointed rooms with all the necessary amenities. Basic amenities offered by Sunkissed Plaza are internet, 24-hour front desk, 24-hour room service, travel desk, 24-hour security and doctor on call. Business services and conference suites are available at Sunkissed Plaza for corporate travellers. There is a swimming pool for a refreshing stay. Wedding and banquet services can be arranged by Sunkissed Plaza if the patrons wish to organise social events. Game lovers can step inside the game room and spend their leisure time. For ultimate rejuvenation and relaxation, patrons can avail of Jacuzzi. Patrons can also sit back and relax at the lounge during the stay. The multi-cuisine restaurant serves tempting meals that gratify the hunger cravings of the guests.\r\n\r\n<b>Rooms:</b>\r\nStandard, premium room and deluxe rooms are available for lodging and boarding. The rooms are cosy with a modern feel to assure an undisturbed stay at Sunkissed Plaza. Elegantly appointed, these room offer a beautiful view of the lush greenery and swimming pool. Air conditioning, colour television, minibar, refrigerator, safe and telephone are the room amenities provided at Sunkissed Plaza.',
  hotelImages: [
    {
      id: 412558,
      url: 'https://res.cloudinary.com/dma3zc8eo/image/fetch/f_auto,q_auto:eco,w_630/https://dknfdeb8v69ms.cloudfront.net/attachments/pictures/412558/original/5296187_21_z.jpg'
    },
    {
      id: 412570,
      url: 'https://res.cloudinary.com/dma3zc8eo/image/fetch/f_auto,q_auto:eco,w_630/https://dknfdeb8v69ms.cloudfront.net/attachments/pictures/412570/original/5296187_22_z.jpg'
    },
    {
      id: 412585,
      url: 'https://res.cloudinary.com/dma3zc8eo/image/fetch/f_auto,q_auto:eco,w_630/https://dknfdeb8v69ms.cloudfront.net/attachments/pictures/412585/original/5296187_31_z.jpg'
    },
    {
      id: 412594,
      url: 'https://res.cloudinary.com/dma3zc8eo/image/fetch/f_auto,q_auto:eco,w_630/https://dknfdeb8v69ms.cloudfront.net/attachments/pictures/412594/original/5296187_35_z.jpg'
    },
    {
      id: 412604,
      url: 'https://res.cloudinary.com/dma3zc8eo/image/fetch/f_auto,q_auto:eco,w_630/https://dknfdeb8v69ms.cloudfront.net/attachments/pictures/412604/original/5296187_36_z.jpg'
    },
    {
      id: 412617,
      url: 'https://res.cloudinary.com/dma3zc8eo/image/fetch/f_auto,q_auto:eco,w_630/https://dknfdeb8v69ms.cloudfront.net/attachments/pictures/412617/original/5296187_26_z.jpg'
    },
    {
      id: 412627,
      url: 'https://res.cloudinary.com/dma3zc8eo/image/fetch/f_auto,q_auto:eco,w_630/https://dknfdeb8v69ms.cloudfront.net/attachments/pictures/412627/original/5296187_32_z.jpg'
    },
    {
      id: 412639,
      url: 'https://res.cloudinary.com/dma3zc8eo/image/fetch/f_auto,q_auto:eco,w_630/https://dknfdeb8v69ms.cloudfront.net/attachments/pictures/412639/original/5296187_34_z.jpg'
    },
    {
      id: 412650,
      url: 'https://res.cloudinary.com/dma3zc8eo/image/fetch/f_auto,q_auto:eco,w_630/https://dknfdeb8v69ms.cloudfront.net/attachments/pictures/412650/original/5296187_20_z.jpg'
    },
    {
      id: 412663,
      url: 'https://res.cloudinary.com/dma3zc8eo/image/fetch/f_auto,q_auto:eco,w_630/https://dknfdeb8v69ms.cloudfront.net/attachments/pictures/412663/original/5296187_28_z.jpg'
    }
  ]
};

export const hotel = {
  days: [
    2
  ],
  hotelInfos: [
    hotelInfo
  ]
};

export const hotels = [
  hotel
];
