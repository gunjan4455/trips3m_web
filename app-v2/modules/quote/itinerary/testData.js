export const itineraryInfo = [
  {
    "0": {
      order: '1',
      title: 'Private Transfer from Airport to Hotel',
      description:
        'A representative will meet and take you to you to your hotel.',
      eventTime: '15:35 - India Standard Time - 2 hours',
      sectionType: 'cab',
      imagesUrl: [
        'https://dailystormer.name/wp-content/uploads/2013/07/London-taxi.jpg',
        'https://dailystormer.name/wp-content/uploads/2013/07/London-taxi.jpg'
      ]
    },
    "1": {
      order: '4',
      title: 'Arrival at Dubai Airport.',
      description:
        'Arrival at Dubai Airport at Evening, meet and greet at Airport and we transfer you toyour hotel. Check in to hotel.',
      eventTime: 'Monday, 24 August, 2019',
      sectionType: 'itinerary',
      imagesUrl: []
    },
    "2": {
      order: '2',
      title: 'hotel details',
      description: 'hotel detail',
      eventTime: 'hotel timing',
      sectionType: 'hotel',
      imagesUrl: [
        'https://dailystormer.name/wp-content/uploads/2013/07/London-taxi.jpg'
      ],
      otherDetails: {
        name: 'hotel',
        rating: 5,
        email: '',
        phone: '44445454',
        web_url: '',
        category: '',
        hotelDesc: 'hotel description',
        tripAdvisorId: '',
        lat: '28.489743',
        long: '77.074602'
      }
    },
    "3": {
      order: '3',
      title: 'Night Safari',
      description:
        "At night, you will be taken for a memorable Night Safari experience. Being the first wildlife night park, it shows you nocturnal animalsâ€™ behavior in their natural habitat. This open-air zoo is known to house nearly 130 species of animals. After an exceptional experience, return to your hotel and stay overnight. Tip: Don't feed animals at the Night Safari",
      eventTime: '15:35 - India Standard Time',
      sectionType: 'attraction',
      imagesUrl: []
    },

    "4": {
      order: '4',
      title: 'Arrival',
      description:
        'Arrival at Dubai Airport at Evening, meet and greet at Airport and we transfer you toyour hotel. Chk in to hotel.',
      eventTime: '15:35 - India Standard Time',
      sectionType: 'flight',
      otherDetails: {
        name: 'prashant',
        image_url:
          'https://dailystormer.name/wp-content/uploads/2013/07/London-taxi.jpg',
        address: 'abc',
        rating: 4.3,
        phone: '33434',
        web_url: '',
        airport_des: 'airport description',
        lat: '28.489743',
        long: '77.074602'
      }
    },
    "5": {
      order: '3',
      title: 'Back to Hotel',
      description: 'going back to hotel after sight seeing',
      eventTime: '15:35 - India Standard Time',
      sectionType: 'attraction',
      imagesUrl: []
    }
  }
];
