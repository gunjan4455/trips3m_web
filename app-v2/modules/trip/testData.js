export const tripsData = {
  loaded: true,
  loading: false,
  allIds: [
    '4809460',
    '4982568',
    '4982590',
    '4982978'
  ],
  byId: {
    '4809460': {
      id: 4809460,
      tripDays: 5,
      toLoc: 'Sri Lanka',
      firstDestinationName: 'Sri Lanka',
      startDate: '2019-07-04',
      quotesCount: 3,
      bookingStatusLabel: 'Incomplete',
      exclusivePreferencesValues: {
        cabs: 'Included',
        flights: 'Included',
        hotels: '4 Star'
      },
      budget: '38000..41000',
      fromLoc: 'Chennai, Tamil Nadu, India',
      info: '',
      adult: 2,
      status: 'Cancelled',
      child: 0,
      tripStage: 3,
      currency: 'Rupee',
      phoneNo: '+919480946000'
    },
    '4982568': {
      id: 4982568,
      tripDays: 9,
      toLoc: 'Singapore',
      firstDestinationName: 'Singapore',
      startDate: '2019-08-11',
      quotesCount: 0,
      bookingStatusLabel: 'Incomplete',
      exclusivePreferencesValues: {
        cabs: 'Included',
        flights: 'Included',
        hotels: '3 Star, 2 Star'
      },
      budget: null,
      fromLoc: 'Chennai, Tamil Nadu, India',
      info: '',
      adult: 2,
      status: 'Dormant',
      child: 0,
      tripStage: 2,
      currency: null,
      phoneNo: '+919498256800'
    },
    '4982590': {
      id: 4982590,
      tripDays: 7,
      toLoc: 'Singapore',
      firstDestinationName: 'Singapore',
      startDate: '2019-08-19',
      quotesCount: 0,
      bookingStatusLabel: 'Incomplete',
      exclusivePreferencesValues: {
        cabs: 'Included',
        flights: 'Excluded',
        hotels: '3 Star'
      },
      budget: '33500..33500',
      fromLoc: 'Chennai',
      info: '',
      adult: 4,
      status: 'Dormant',
      child: 0,
      tripStage: 3,
      currency: 'Rupee',
      phoneNo: '+919498259000'
    },
    '4982978': {
      id: 4982978,
      tripDays: 7,
      toLoc: 'Singapore',
      firstDestinationName: 'Singapore',
      startDate: '2019-08-19',
      quotesCount: 3,
      bookingStatusLabel: 'Active',
      exclusivePreferencesValues: {
        cabs: 'Included',
        flights: 'Excluded',
        hotels: '3 Star'
      },
      budget: '33500..33500',
      fromLoc: 'Chennai',
      info: '',
      adult: 4,
      status: 'Active',
      child: 0,
      tripStage: 3,
      currency: 'Rupee',
      phoneNo: '+919498297800'
    }
  }
};