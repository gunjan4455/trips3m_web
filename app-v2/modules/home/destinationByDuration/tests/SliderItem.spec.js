import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import SlideItem from '../SlideItem';
import { slideItem } from './testData';

configure({ adapter: new Adapter() });

const shallowWrappperSlideItem = (props) => {
  return shallow(<SlideItem {...props} />);
};

const DEFAULT_PROPS = {
  ...slideItem
};

describe('SlideItem: Render Correctly ', () => {

  let props = { ...DEFAULT_PROPS };

  afterEach(() => {
    props = { ...DEFAULT_PROPS };
  });

  it('should match with the previous snapshot', () => {
    const wrapper = shallowWrappperSlideItem(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not have wMaxFull class when packageCount is 0', () => {
    props.packageCount = 0;
    const wrapper = shallowWrappperSlideItem(props);

    expect(wrapper.find('.at_budgetlistbox')).toHaveLength(0);
  });

});
