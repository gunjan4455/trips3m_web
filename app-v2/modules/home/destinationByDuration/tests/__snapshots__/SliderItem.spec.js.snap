// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlideItem: Render Correctly  should match with the previous snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SlideItem
    bestTimeToVisit={
      Array [
        2,
        5,
      ]
    }
    id={60}
    imgPath="https://res.cloudinary.com/dxqin8fbx/image/fetch/f_auto,q_auto:eco,w_148,h_187/https://d29m6q8e2acg44.cloudfront.net/public-product/Destination_Module/Generic/Domestic/Kerala_shutterstock_493080757_2x.jpg"
    link="/tour-packages/andaman"
    name="Andaman"
    packageCount={104}
    startingPrice={4000}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        onClick={[Function]}
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/tour-packages/andaman"
      >
        <div
          className="radius4 overflowh"
          style={
            Object {
              "height": "150px",
              "width": "120px",
            }
          }
        >
          <Img
            alt="Andaman"
            defaultSrc={null}
            height={150}
            onDocumentLoad={false}
            width={120}
          />
        </div>
        <VisibilitySensor
          active={true}
          containment={null}
          delayedCall={false}
          intervalCheck={true}
          intervalDelay={3000}
          minTopValue={0}
          offset={Object {}}
          onChange={[Function]}
          partialVisibility={false}
          resizeCheck={false}
          resizeDelay={250}
          resizeThrottle={-1}
          scrollCheck={false}
          scrollDelay={250}
          scrollThrottle={-1}
        >
          <span />
        </VisibilitySensor>
        <p
          className="f14p fw9 pfc3 mt8"
          style={
            Object {
              "width": "120px",
            }
          }
        >
          Andaman
        </p>
        <p
          className="f10 pfc3 mt4"
        >
          Starting at: 
          ₹4,000
        </p>
      </Link>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="radius4 overflowh"
            style={
              Object {
                "height": "150px",
                "width": "120px",
              }
            }
          >
            <Img
              alt="Andaman"
              defaultSrc={null}
              height={150}
              onDocumentLoad={false}
              width={120}
            />
          </div>,
          <VisibilitySensor
            active={true}
            containment={null}
            delayedCall={false}
            intervalCheck={true}
            intervalDelay={3000}
            minTopValue={0}
            offset={Object {}}
            onChange={[Function]}
            partialVisibility={false}
            resizeCheck={false}
            resizeDelay={250}
            resizeThrottle={-1}
            scrollCheck={false}
            scrollDelay={250}
            scrollThrottle={-1}
          >
            <span />
          </VisibilitySensor>,
          <p
            className="f14p fw9 pfc3 mt8"
            style={
              Object {
                "width": "120px",
              }
            }
          >
            Andaman
          </p>,
          <p
            className="f10 pfc3 mt4"
          >
            Starting at: 
            ₹4,000
          </p>,
        ],
        "onClick": [Function],
        "onlyActiveOnIndex": false,
        "style": Object {},
        "to": "/tour-packages/andaman",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Img
              alt="Andaman"
              defaultSrc={null}
              height={150}
              onDocumentLoad={false}
              width={120}
            />,
            "className": "radius4 overflowh",
            "style": Object {
              "height": "150px",
              "width": "120px",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "Andaman",
              "defaultSrc": null,
              "height": 150,
              "ikSrc": undefined,
              "onDocumentLoad": false,
              "src": undefined,
              "width": 120,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": true,
            "children": <span />,
            "containment": null,
            "delayedCall": false,
            "intervalCheck": true,
            "intervalDelay": 3000,
            "minTopValue": 0,
            "offset": Object {},
            "onChange": [Function],
            "partialVisibility": false,
            "resizeCheck": false,
            "resizeDelay": 250,
            "resizeThrottle": -1,
            "scrollCheck": false,
            "scrollDelay": 250,
            "scrollThrottle": -1,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Andaman",
            "className": "f14p fw9 pfc3 mt8",
            "style": Object {
              "width": "120px",
            },
          },
          "ref": null,
          "rendered": "Andaman",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Starting at: ",
              "₹4,000",
            ],
            "className": "f10 pfc3 mt4",
          },
          "ref": null,
          "rendered": Array [
            "Starting at: ",
            "₹4,000",
          ],
          "type": "p",
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          onClick={[Function]}
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/tour-packages/andaman"
        >
          <div
            className="radius4 overflowh"
            style={
              Object {
                "height": "150px",
                "width": "120px",
              }
            }
          >
            <Img
              alt="Andaman"
              defaultSrc={null}
              height={150}
              onDocumentLoad={false}
              width={120}
            />
          </div>
          <VisibilitySensor
            active={true}
            containment={null}
            delayedCall={false}
            intervalCheck={true}
            intervalDelay={3000}
            minTopValue={0}
            offset={Object {}}
            onChange={[Function]}
            partialVisibility={false}
            resizeCheck={false}
            resizeDelay={250}
            resizeThrottle={-1}
            scrollCheck={false}
            scrollDelay={250}
            scrollThrottle={-1}
          >
            <span />
          </VisibilitySensor>
          <p
            className="f14p fw9 pfc3 mt8"
            style={
              Object {
                "width": "120px",
              }
            }
          >
            Andaman
          </p>
          <p
            className="f10 pfc3 mt4"
          >
            Starting at: 
            ₹4,000
          </p>
        </Link>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="radius4 overflowh"
              style={
                Object {
                  "height": "150px",
                  "width": "120px",
                }
              }
            >
              <Img
                alt="Andaman"
                defaultSrc={null}
                height={150}
                onDocumentLoad={false}
                width={120}
              />
            </div>,
            <VisibilitySensor
              active={true}
              containment={null}
              delayedCall={false}
              intervalCheck={true}
              intervalDelay={3000}
              minTopValue={0}
              offset={Object {}}
              onChange={[Function]}
              partialVisibility={false}
              resizeCheck={false}
              resizeDelay={250}
              resizeThrottle={-1}
              scrollCheck={false}
              scrollDelay={250}
              scrollThrottle={-1}
            >
              <span />
            </VisibilitySensor>,
            <p
              className="f14p fw9 pfc3 mt8"
              style={
                Object {
                  "width": "120px",
                }
              }
            >
              Andaman
            </p>,
            <p
              className="f10 pfc3 mt4"
            >
              Starting at: 
              ₹4,000
            </p>,
          ],
          "onClick": [Function],
          "onlyActiveOnIndex": false,
          "style": Object {},
          "to": "/tour-packages/andaman",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Img
                alt="Andaman"
                defaultSrc={null}
                height={150}
                onDocumentLoad={false}
                width={120}
              />,
              "className": "radius4 overflowh",
              "style": Object {
                "height": "150px",
                "width": "120px",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "Andaman",
                "defaultSrc": null,
                "height": 150,
                "ikSrc": undefined,
                "onDocumentLoad": false,
                "src": undefined,
                "width": 120,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": true,
              "children": <span />,
              "containment": null,
              "delayedCall": false,
              "intervalCheck": true,
              "intervalDelay": 3000,
              "minTopValue": 0,
              "offset": Object {},
              "onChange": [Function],
              "partialVisibility": false,
              "resizeCheck": false,
              "resizeDelay": 250,
              "resizeThrottle": -1,
              "scrollCheck": false,
              "scrollDelay": 250,
              "scrollThrottle": -1,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Andaman",
              "className": "f14p fw9 pfc3 mt8",
              "style": Object {
                "width": "120px",
              },
            },
            "ref": null,
            "rendered": "Andaman",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Starting at: ",
                "₹4,000",
              ],
              "className": "f10 pfc3 mt4",
            },
            "ref": null,
            "rendered": Array [
              "Starting at: ",
              "₹4,000",
            ],
            "type": "p",
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
